pkg/ingester/ingester_v2.go

```diff
@@ -109,6 +110,38 @@ func (u *userTSDB) PostDeletion(metrics ...labels.Labels) {
 	}
 }
 
+// blocksToDelete returns the blocks which have been shipped.
+func (u *userTSDB) blocksToDelete(blocks []*tsdb.Block) map[ulid.ULID]struct{} {
+	deletable := make(map[ulid.ULID]struct{}, len(blocks))
+	for _, b := range blocks {
+		deletable[b.Meta().ULID] = struct{}{}
+	}
+
+	if u.shipper == nil {
+		return deletable
+	}
+
+	shipperMeta, err := shipper.ReadMetaFile(u.Dir())
+	if err != nil {
+		// If there is any issue with the shipper, we should be conservative and not delete anything.
+		level.Error(util.Logger).Log("msg", "failed to read shipper meta during deletion of blocks", "user", u.userID, "err", err)
+		return nil
```

Comment:
Should we have a metric for this to alert on? Because if this issue persists, queries could skip some data and the disk can run out of space.

Reply:
In our alerts we do alert on disk running out of space. I'd say that + error message is enough.